addons/web/static/tests/core/virtual_hook_tests.js

```diff
@@ -0,0 +1,193 @@
+    assert.containsN(target, ".item", 6);
+    assert.strictEqual(target.querySelector(".item").dataset.id, "98");
+    assert.strictEqual(target.querySelector(".item:last-child").dataset.id, "103");
+
+    // scroll to the end
+    await scroll(target, MAX_SCROLL_TOP);
+    assert.containsN(target, ".item", 6);
+    assert.strictEqual(target.querySelector(".item").dataset.id, "195");
+    assert.strictEqual(target.querySelector(".item:last-child").dataset.id, "200");
+});
+
+QUnit.test("margin: item relative", async (assert) => {
+    await mountTestComponent(target, { margin: 10 * ITEM_HEIGHT });
+    assert.containsN(target, ".item", 16);
+    assert.strictEqual(target.querySelector(".item").dataset.id, "1");
+    assert.strictEqual(target.querySelector(".item:last-child").dataset.id, "16");
+
+    // scroll to the middle
+    await scroll(target, MAX_SCROLL_TOP / 2);
+    assert.containsN(target, ".item", 26);
+    assert.strictEqual(target.querySelector(".item").dataset.id, "88");
+    assert.strictEqual(target.querySelector(".item:last-child").dataset.id, "113");
+
+    // scroll to the end
+    await scroll(target, MAX_SCROLL_TOP);
+    assert.containsN(target, ".item", 16);
+    assert.strictEqual(target.querySelector(".item").dataset.id, "185");
+    assert.strictEqual(target.querySelector(".item:last-child").dataset.id, "200");
+});
+
+QUnit.test("margin: percentage", async (assert) => {
+    await mountTestComponent(target, { margin: "300%" });
+    assert.containsN(target, ".item", 21);
+    assert.strictEqual(target.querySelector(".item").dataset.id, "1");
+    assert.strictEqual(target.querySelector(".item:last-child").dataset.id, "21");
+
+    // scroll to the middle
+    await scroll(target, MAX_SCROLL_TOP / 2);
+    assert.containsN(target, ".item", 36);
+    assert.strictEqual(target.querySelector(".item").dataset.id, "83");
+    assert.strictEqual(target.querySelector(".item:last-child").dataset.id, "118");
+
+    // scroll to the end
+    await scroll(target, MAX_SCROLL_TOP);
+    assert.containsN(target, ".item", 21);
+    assert.strictEqual(target.querySelector(".item").dataset.id, "180");
+    assert.strictEqual(target.querySelector(".item:last-child").dataset.id, "200");
+});
+
+QUnit.test("initialScroll: middle", async (assert) => {
+    const initialScroll = { top: MAX_SCROLL_TOP / 2 };
+    await mountTestComponent(target, { initialScroll });
+    assert.containsN(target, ".item", 16);
+    assert.strictEqual(target.querySelector(".item").dataset.id, "93");
+    assert.strictEqual(target.querySelector(".item:last-child").dataset.id, "108");
+});
+
+QUnit.test("initialScroll: bottom", async (assert) => {
+    const initialScroll = { top: MAX_SCROLL_TOP };
+    await mountTestComponent(target, { initialScroll });
+    assert.containsN(target, ".item", 11);
+    assert.strictEqual(target.querySelector(".item").dataset.id, "190");
+    assert.strictEqual(target.querySelector(".item:last-child").dataset.id, "200");
```

Comment:
make sure those tests are robusts (i.e. doesn't break in a given browser, doesn't break if not zoom=100%, are deterministic...)

Reply:
I checked and they are ;-)